jobs/vqsr.py

```diff
@@ -0,0 +1,1033 @@
+      GatherTranches \\
+      --mode SNP \\
+      {inputs_cmdl} \\
+      --output {j.out_tranches}"""
+    )
+    return j
+
+
+def add_apply_recalibration_step(
+    b: hb.Batch,
+    input_vcf: hb.ResourceGroup,
+    indels_recalibration: hb.ResourceGroup,
+    indels_tranches: hb.ResourceFile,
+    snps_recalibration: hb.ResourceGroup,
+    snps_tranches: hb.ResourceFile,
+    disk_size: int,
+    use_as_annotations: bool,
+    indel_filter_level: float,
+    snp_filter_level: float,
+    interval: hb.Resource | None = None,
+    output_vcf_path: Path | None = None,
+    job_attrs: dict | None = None,
+) -> Job:
+    """
+    Apply a score cutoff to filter variants based on a recalibration table.
+    Runs ApplyVQSR twice to apply first indel, then SNP recalibrations.
+
+    Targets indel_filter_level and snp_filter_level sensitivities. The tool matches
+    them internally to a VQSLOD score cutoff based on the model's estimated sensitivity
+    to a set of true variants.
+
+    The filter determination is not just a pass/fail process. The tool evaluates for
+    each variant which "tranche", or slice of the dataset, it falls into in terms of
+    sensitivity to the truthset. Variants in tranches that fall below the specified
+    truth sensitivity filter level have their FILTER field annotated with the
+    corresponding tranche level. This results in a callset that is filtered to the
+    desired level but retains the information necessary to increase sensitivity
+    if needed.
+
+    Returns: a Job object with one ResourceGroup output j.output_vcf, corresponding
+    to a VCF with tranche annotated in the FILTER field
+    """
+    j = b.new_job('VQSR: ApplyVQSR', job_attrs)
+    j.image(image_path('gatk'))
+    res = STANDARD.set_resources(j, ncpu=2, storage_gb=disk_size)
+
+    j.declare_resource_group(
+        output_vcf={'vcf.gz': '{root}.vcf.gz', 'vcf.gz.tbi': '{root}.vcf.gz.tbi'}
+    )
+
+    assert isinstance(j.output_vcf, hb.ResourceGroup)
+    j.command(
+        f"""set -euo pipefail
+
+    df -h; pwd; du -sh $(dirname {j.output_vcf['vcf.gz']})
+
+    TMP_DIR=$(dirname {j.output_vcf['vcf.gz']})/tmp
+    mkdir $TMP_DIR
+
+    TMP_INDEL_RECALIBRATED=$BATCH_TMPDIR/tmp.indel.recalibrated.vcf.gz
+    gatk --java-options -Xms{res.get_java_mem_mb()}m \\
+      ApplyVQSR \\
+      --tmp-dir $TMP_DIR \\
+      -O $TMP_INDEL_RECALIBRATED \\
+      -V {input_vcf['vcf.gz']} \\
+      --recal-file {indels_recalibration} \\
+      --tranches-file {indels_tranches} \\
+      --truth-sensitivity-filter-level {indel_filter_level} \\
+      --create-output-variant-index true \\
+      {f'-L {interval} ' if interval else ''} \\
+      {'--use-allele-specific-annotations ' if use_as_annotations else ''} \\
+      -mode INDEL
+      
+    df -h; pwd; du -sh $(dirname {j.output_vcf['vcf.gz']})
+
+    rm {input_vcf['vcf.gz']} {indels_recalibration} {indels_tranches}
+    rm -rf $TMP_DIR
+    mkdir $TMP_DIR
```

Comment:
just being pedantic, but wouldn't this save a command?
```suggestion
    rm -rf $TMP_DIR/*
```